src/include/execution/sql/runtime_types.h

```diff
@@ -488,9 +513,101 @@ class EXPORT Decimal {
     return *this;
   }
 
+  /**
+   * Get the underlying value of the decimal
+   * @return This decimal value.
+   */
+  T GetValue() const { return value_; }
+
+  /**
+   * Set the underlying value of the decimal
+   * @param value The value to be set.
+   */
+  void SetValue(T value) { value_ = value; }
+
+  /** This function divides with a given decimal
+   * Note we multiply with the 10^precision of the denominator and then
+   * divide by the denominator
+   * If we do not overflow, we use a 128 number division
+   * If we already know the magic number of the 128 bit number then we use
+   * magic number division
+   * If we overflow, we need to use a multiword 256 bit division
+   * If we already know the magic number of the 256 bit number then we use
+   * magic number division
+   * We retain the result in numerator precision, this is due to the simple
+   * fact that if we were to retain it in denominator precision we need to
+   * multiply with 10 ^ (2* denominator - numerator) which will require a
+   * 256 x 256 bit multiplication with an overflow check in 512 bits
+   * @param input the decimal to be divide with
+   * @param denominator_precision Number of digits after decimal point.*/
+  void SignedDivideWithDecimal(Decimal<T> input, uint32_t denominator_precision);
+
+  /** This function divides with a given unsigned 128 bit number with
+   * another 128 bit input. The given number here is stored inside the
+   * decimal class and assumes that it is non negative.
+   * If constant is a power of 2, we right shift
```

Comment:
Nit: this is all implementation details right? Does the caller need to know any of this? If no, I think the comment should move to the implementation then.

Reply:
going to stop copy pasting, this generally applies throughout the file